docs/lincheck-test-tutorial.md

```diff
@@ -91,41 +97,38 @@ class CounterTest {
+   fun test() = ModelCheckingOptions().check(this::class)
 }
 ```
 
-Now you see the whole execution trace leading to the incorrect results:
+When you re-run the test with model checking instead of stress testing, you will get the execution trace leading to the incorrect results:
     
 ```text
 = Invalid execution results =
 Parallel part:
 | inc(): 1 | inc(): 1 |
 = The following interleaving leads to the error =
 Parallel part trace:
-|                      | inc()                                                        |
-|                      |   inc(): 1 at CounterTest.inc(CounterTest.kt:146) |
-|                      |     value.READ: 0 at Counter.inc(CounterTest.kt:105)           |
-|                      |     switch                                                   |
-| inc(): 1             |                                                              |
-|   thread is finished |                                                              |
-|                      |     value.WRITE(1) at Counter.inc(CounterTest.kt:105)          |
-|                      |     value.READ: 1 at Counter.inc(CounterTest.kt:105)           |
-|                      |   result: 1                                                  |
-|                      |   thread is finished                                         |
+|                      | inc()                                                 |
+|                      |   inc(): 1 at CounterTest.inc(CounterTest.kt:146)     |
+|                      |     value.READ: 0 at Counter.inc(CounterTest.kt:105)  |
+|                      |     switch                                            |
+| inc(): 1             |                                                       |
+|   thread is finished |                                                       |
+|                      |     value.WRITE(1) at Counter.inc(CounterTest.kt:105) |
+|                      |     value.READ: 1 at Counter.inc(CounterTest.kt:105)  |
+|                      |   result: 1                                           |
+|                      |   thread is finished                                  |
 ```
     
-Let's read the trace:
-
-1. Thread 2 read the current value of the counter `value.READ: 0`
-2. The execution switched to Thread 1
-3. Thread 1 executed `inc()` completely and returned `1`
-4. The execution switched back to the Thread 2
-5. Thread 2 increments the old value of the counter that was read before the switch and returned `1`
-
+In this trace, the following list of events have occurred:  
+**T2:** The 2nd thread reads the current value of the counter (`value.READ: 0`) and pauses.    
+**T1:** The 1st thread executes `inc()`, which returns `1`, and finishes.  
+**T2:** The 2nd thread increments the previously read value of the counter and incorrectly updates it to `1`, returning `1` as a result.
 
 ## Test structure
+TODO: I am not sure that we should discuss the test structure in the "Lincheck Hello World" section.
```

Comment:
Maybe, you're right, no need to overload the intro. I'll move the explanation of the structure to the next section then